test/TestFullIntegration.cxx

```diff
@@ -0,0 +1,974 @@
+		conf_file << "    type \"null\"\n";
+		conf_file << "    name \"MyTestOutput\"\n";
+		conf_file << "    mixer_type \"null\"\n";
+		conf_file << "}\n";
+
+		// Add any extra configuration
+		for (const auto& line : extra_config) {
+			conf_file << line << "\n";
+		}
+
+		if (!conf_file) {
+			throw std::runtime_error("Failed to write config file");
+		}
+	}
+
+	/**
+	 * Start the MPD process.
+	 * Forks a child process and executes MPD with the test configuration.
+	 *
+	 * @param startup_delay_ms Time to wait for MPD to start (milliseconds)
+	 * @return true if MPD was started successfully, false otherwise
+	 */
+	[[nodiscard]] bool StartMpd(int startup_delay_ms = kDefaultStartupDelayMs) {
+		mpd_pid_ = fork();
+
+		if (mpd_pid_ == 0) {
+			// Child process - execute MPD
+			std::vector<char> mpd_exec_buf(mpd_executable_.begin(), mpd_executable_.end());
+			mpd_exec_buf.push_back('\0');
+
+			std::string conf_str = conf_path_.string();
+			std::vector<char> conf_buf(conf_str.begin(), conf_str.end());
+			conf_buf.push_back('\0');
+
+			char* argv[] = { mpd_exec_buf.data(), conf_buf.data(), nullptr };
+			execv(mpd_exec_buf.data(), argv);
+
+			// If we get here, execv failed
+			perror("execv failed");
+			exit(1);
+		}
+
+		if (mpd_pid_ < 0) {
+			return false;
+		}
+
+		// Poll for MPD to start
+		// Use a polling interval of 10ms, retrying until startup_delay_ms is reached
+		for (int i = 0; i < startup_delay_ms / 10; ++i) {
+			if (Connect()) {
+				return true;
+			}
+			std::this_thread::sleep_for(std::chrono::milliseconds(10));
+		}
+		return false; // MPD did not start within the timeout
+	}
+
+	/**
+	 * Connect to the MPD socket.
+	 * Establishes a Unix domain socket connection and reads the greeting.
+	 *
+	 * @return true if connection succeeded, false otherwise
+	 */
+	[[nodiscard]] bool Connect() {
+		sock_ = socket(AF_UNIX, SOCK_STREAM, 0);
+		if (sock_ < 0) {
+			perror("socket failed");
+			return false;
+		}
+
+		struct sockaddr_un addr{};
+		addr.sun_family = AF_UNIX;
+		std::strncpy(addr.sun_path, socket_path_.c_str(), sizeof(addr.sun_path) - 1);
+
+		if (::connect(sock_, reinterpret_cast<struct sockaddr*>(&addr), sizeof(addr)) < 0) {
+			close(sock_);
+			sock_ = -1;
+			return false;
+		}
+
+		// Read and display greeting
+		char buffer[kGreetingBufferSize];
+		ssize_t bytes_read = recv(sock_, buffer, sizeof(buffer) - 1, 0);
+		if (bytes_read > 0) {
+			buffer[bytes_read] = '\0';
+			std::cout << "MPD greeting: " << buffer;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Send a command to MPD and receive the response.
+	 * Automatically appends newline if not present.
+	 *
+	 * @param command The MPD protocol command to send
+	 * @return The response from MPD, or error message if not connected
+	 */
+	[[nodiscard]] std::string SendCommand(const std::string& command) {
+		if (sock_ < 0) {
+			return "ERROR: Not connected";
+		}
+
+		std::string cmd = command;
+		if (cmd.empty() || cmd.back() != '\n') {
```

Comment:
If the command is empty, send a bare newline character? What's the point of that?